### PR TITLE
fix: add spam filtering for lists (closes #11905)

### LINKED ISSUE #11905: Remove Spam Lists
### Problem

The top results for this search are all spam: https://openlibrary.org/search/lists?q=best
covering topics such as oxycontin, bookies, makeup academies, etc.

### Reproducing the bug

1. Go to https://openlibrary.org/search/lists?q=best
2. Do examine list

* Expected behavior: List descriptions should be scored on their spaminess and either blocked or sent to curation if the score is too high. 
* Actual behavior: Spam lists are allowed 

### Context

- Browser (Chrome, Safari, Firefox, etc):
- OS (Windows, Mac, etc):
- Logged in (Y/N):
- Environment (prod, dev, local): prod

### Breakdown

<details>
<summary><strong>Implementation Details</strong> (for maintainers)</summary>

#### Requirements Checklist
* [ ]

#### Related files

*

#### Stakeholders

*
<hr>

#### Instructions for Contributors

- Please [run these commands](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#working-on-your-branch) to ensure your repository is up to date **before** [creating a new branch](https://github.com/internetarchive/openlibrary/wiki/Git-Cheat-Sheet#making-changes-and-creating-a-pull-request) to work on this issue and **each time after** pushing code to Github, because the pre-commit bot may add commits to your PRs upstream.

</details>

### PR DESCRIPTION
**Pull request by shoaib-inamdar**
- Block zero-seed lists from Solr indexing (87% of spam has 0 seeds)
- Add is_spam_list() to check list names AND descriptions against spam words
- Add spam guard in lists_edit.POST() to block spam via HTML form
- Add Solr fq filters to exclude zero-seed and spam-named lists from search
- Add cleanup script for existing spam lists (scripts/cleanup_spam_lists.py)
- Add 7 new tests (5 for is_spam_list, 2 for zero-seed Solr behavior)

Based on analysis of 254,050 lists from ol_dump_lists dump:
- 23,091 zero-seed lists (9.1%)
- 13,740 high-confidence spam (pattern match + 0 seeds)
- Top categories: casino/gambling, pharmaceuticals, phone scams

Closes #11905 

Adds multi-layered spam filtering for Open Library lists, informed by analysis of 254,050 lists from the `ol_dump_lists` database dump.

### Testing
- All **17 focused tests pass** in Docker
- Full CI-style test suite: **2940 passed**, 16 skipped, 0 failures
- Lint (ruff): clean on all modified files

### CODE REVIEW

File: openlibrary/plugins/upstream/spamcheck.py [hunk 1]
```diff
@@ -63,3 +63,19 @@ def is_spam(i=None, allow_privileged_edits: bool = False) -> bool:
 def is_spam_email(email: str) -> bool:
     domain = email.rsplit("@", maxsplit=1)[-1].lower()
     return domain in get_spam_domains()
+
+
+# NEW FUNCTION — catches spam in list names AND descriptions
+def is_spam_list(name: str, description: str = "") -> bool:
```
Comment (cdrini) [inline openlibrary/plugins/upstream/spamcheck.py]: We can use the existing `is_spam` method ; it'll check all fields in the provided first object. You should be able to do `is_spam(list_record.to_thing_json())`

File: openlibrary/plugins/worksearch/schemes/lists.py [hunk 2]
```diff
@@ -66,8 +66,20 @@ def q_to_solr_params(
...
+        # Exclude lists whose names match spam words
+        if spam_words := get_spam_words():
+            # Escape Solr special chars in each word to prevent query injection
+            import re as _re
+
+            _solr_special = _re.compile(r'([+\-&|!(){}[\]^"~*?:\\/])')
+            safe_words = [_solr_special.sub(r'\\\1', w) for w in spam_words]
+            escaped = ' OR '.join(f'"{w}"' for w in safe_words)
+            params.append(('fq', f'-name:({escaped})'))
+        return params
```
Comment (cdrini) [inline openlibrary/plugins/worksearch/schemes/lists.py]: Preventing spam lists from being created is sufficient. Since we will need to run a cleanup step to delete all those spam lists, we don't want to add this kind of logic to the queries.

File: openlibrary/solr/updater/list.py [hunk 3]
```diff
@@ -23,7 +23,16 @@ def key_test(self, key: str) -> bool:
...
+        builder = ListSolrBuilder(list)
+
+        # NEW: Lists with 0 seeds should not be indexed.
+        # Our dump analysis found 23,091 zero-seed lists (9.1% of all lists),
+        # and 87% of detected spam falls in this category.
+        # Only apply to lists, not series (which may intentionally have 0 seeds).
+        if not builder.seed and not builder.key.startswith('/series/'):
+            return SolrUpdateRequest(deletes=[builder.key]), []
+
+        seeds = builder.seed
```
Comment (cdrini) [inline openlibrary/solr/updater/list.py]: Ditto here ; the preventing save is sufficient. This can be rolled back.

File: openlibrary/plugins/worksearch/schemes/lists.py [hunk 4]
```diff
@@ -66,8 +66,20 @@ def q_to_solr_params(
...
-        return [
-            ('q', q),  # actual query string
-            ('q.op', 'AND'),  # use 'AND" for matching multiple words in search queries
-            ('defType', 'edismax'),  # use edismax parser for better full-text search
+        params = [
+            ('q', q),
+            ('q.op', 'AND'),
+            ('defType', 'edismax'),
+            # Exclude zero-seed lists (23,091 in dump, 87% of spam)
+            ('fq', 'seed_count:[1 TO *]'),
```
Comment (cdrini) [inline openlibrary/plugins/worksearch/schemes/lists.py]: Actually, let's set this _only if_ `seed_count` isn't in `q`. That way we can override it.

Also, let's make this: `seed_count:[2 TO *] OR list_type:series`. User lists with only one book aren't much of a list! But there are some series, eg new series, that only have one book on them.

**Comment by shoaib-inamdar** [comment]
@cdrini , aah sorry, i just got excited and implemented all the changes 😅.
